Packs/CiscoUmbrellaReporting/Integrations/CiscoUmbrellaReporting/CiscoUmbrellaReporting.py

```diff
@@ -0,0 +1,1317 @@
+            "threat_types", "amp_disposition", "page", "page_size", "ports"],
+    "category": ["summary_type", "limit", "from", "to", "offset", "domains", "urls", "ip",
+                 "identity_types", "verdict", "file_name", "threats",
+                 "threat_types", "amp_disposition", "page", "page_size"],
+    "destination": ["summary_type", "limit", "from", "to", "offset", "domains", "urls", "ip",
+                    "identity_types", "verdict", "file_name", "threats",
+                    "threat_types", "amp_disposition", "page", "page_size"],
+    "intrusion_rule": ["summary_type", "limit", "from", "to", "offset", "signatures", "ip",
+                       "identity_types", "intrusion_action", "ports", "page",
+                       "page_size"]
+}
+''' CLIENT CLASS '''
+
+
+class Client(BaseClient):
+    """
+    This Client implements API calls, and does not contain any XSOAR logic.
+    Should only do requests and return data.
+    It inherits from BaseClient defined in CommonServer Python.
+    Most calls use _http_request() that handles proxy, SSL verification, etc.
+    For this implementation, no special attributes defined
+    """
+    def __init__(self, base_url: str, organisation_id: str,
+                 secret_key: str, client_key: str,
+                 verify=None,
+                 proxy=None):
+        super().__init__(
+            base_url,
+            verify=verify,
+            proxy=proxy,
+            ok_codes=(200,),
+        )
+        self.token_url = TOKEN_ENDPOINT
+        self.secret_key = secret_key
+        self.client_key = client_key
+        self.organisation_id = organisation_id
+        if not self.organisation_id.isdigit():
+            raise DemistoException("Invalid Input Error: The Organization ID must be a number.")
+
+    def access_token(self):
+        """
+        Generate Access token
+        Returns:
+            Returns the access_token
+        """
+        payload = {
+            "grant_type": 'client_credentials'
+        }
+
+        token_response = requests.post(url=self.token_url, auth=HTTPBasicAuth(
+            username=self.client_key, password=self.secret_key), data=payload)
+        if token_response.status_code == 401:
+            raise DemistoException(INVALID_CREDENTIALS_ERROR_MSG)
+        elif token_response.status_code >= 400:
+            raise DemistoException('Error: something went wrong, please try again.')
+
+        return token_response.json().get('access_token')
+
+    def fetch_data_from_cisco_api(self, end_point: str, params: dict) -> Dict:
+        """
+        Fetch Cisco Umbrella Reporting API
+        Args:
+            end_point (str): Cisco Umbrella Reporting endpoint
+            params (dict): Params
+        Returns:
+             Return the raw api response from Cisco Umbrella Reporting API.
+        """
+        return self.query(end_point, params)
+
+    def query(self, end_point: str, params: dict) -> Dict:
```

Comment:
I noticed that the 'fetch_data_from_cisco_api' function only calls to the 'query' function... why do we need them both?

Reply:
@ShacharKidor removed